### DISCUSSION TITLE
Upload a file via POST request

### DISCUSSION THREAD

**Discussion by savornicesei** [warmuuh/milkman#190] [Q&A] [answered]
Hi all,

How could I upload a file via POST request in Milkman?
Postman provides lots of [such examples](https://www.postman.com/postman/workspace/postman-answers/documentation/13455110-00378d5c-5b08-4813-98da-bc47a2e6021d) but I'm not sure how to translate them in Milkman.

Thank you.

**Comment by warmuuh** [comment, marked as answer]
handling (external) files is currently not supported, see https://github.com/warmuuh/milkman/issues/134
there might be a plugin in the future to load files such as `{{file:path-to-file}}` but for now, you can help yourself with a script:
create a script-file like this somewhere:
```javascript
function loadFile(path) {
 var pathObj = java.nio.file.Paths.get(path)
 var bytesObj = java.nio.file.Files.readAllBytes(pathObj);
 var bytes = Java.from(bytesObj) // converting to JavaScript
 return String.fromCharCode.apply(null, bytes)
}
```

and add the file in the scripting-options like `file:///path/to/loadfile.js`


then, you can include your file for example like this in the body of a request:
```json
{
  "csvFile": "{{js:base64(loadFile('/path/to/test.csv'))}}"
}
```
